scene/main/window.cpp

```diff
@@ -677,16 +677,24 @@ void Window::_event_callback(DisplayServer::WindowEvent p_event) {
 	switch (p_event) {
 		case DisplayServer::WINDOW_EVENT_MOUSE_ENTER: {
 			_propagate_window_notification(this, NOTIFICATION_WM_MOUSE_ENTER);
-			emit_signal(SNAME("mouse_entered"));
+			Window *root = get_tree()->get_root();
+#ifdef DEBUG_ENABLED
+			ERR_FAIL_COND_MSG(root->gui.windowmanager_window_over, "Entering a window while a window is hovered should never happen.");
+#endif // DEBUG_ENABLED
```

Comment:
If that should never happen, as in "this is an engine bug," maybe a `DEV_ASSERT` is better (remember the condition has to be reversed for that macro).

Reply:
Indeed this was intended for engine bugs. I agree, that `DEV_ASSERT` is a better fit.